packages/solid-router/src/HeadContent.tsx

```diff
@@ -5,6 +5,33 @@ import { useRouter } from './useRouter'
 import { useRouterState } from './useRouterState'
 import type { RouterManagedTag } from '@tanstack/router-core'
 
+const isTruthy = (val?: string | boolean) => val === '' || val === true
```

Comment:
we should move the whole sorting logic into router-core
also add a comment please where that is originating from

Reply:
thinking about this more, maybe we should just expose an API that a user could hook into to apply sorting using existing libraries instead of copying the stuff into router.

Reply:
> also add a comment please where that is originating from

Can you clarify what you mean?